File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    dic ={}
    answer = []
    for idx in range(len(name)):
        dic[name[idx]]=yearning[idx]

    for i in photo:
        score=0
        for j in i:
            score+=dic.get(j,0)
        answer.append(score)
    
    return answer

File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    const score = {};
    let answer = [];
    for(let i=0; i < name.length; i++){
        score[name[i]] = yearning[i];
    };
    
    for(arr of photo){
        let total = 0;
        for(val of arr){
            if(score[val] !== undefined) total+=score[val];
        };
        answer.push(total);
    };
    
    return answer;
}